### PR TITLE
Renamed Mount root_dir Argument to mount_path

### PR DESCRIPTION
**Pull request by rlizzo**
## What does this PR do?

At the request of @edenafek and @awaelchli we are renaming the `Mount()` argument `root_dir -> mount_path`. As these changes are unreleased this has no breaking change implications. 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning_app/storage/mount.py [hunk 1]
```diff
@@ -42,8 +43,14 @@ def __post_init__(self) -> None:
                 f"Received: '{self.source}'. Mounting a single file is not currently supported."
             )
 
-        if self.root_dir == "":
-            self.root_dir = f"/data/{Path(self.source).stem}"
+        if self.mount_path == "":
+            self.mount_path = f"/data/{Path(self.source).stem}"
+
+        if not os.path.isabs(self.mount_path):
```
Comment (lantiga) [inline src/lightning_app/storage/mount.py:49]: good call, maybe add a test for this condition?

File: src/lightning_app/utilities/packaging/cloud_compute.py [hunk 2]
```diff
@@ -141,9 +141,9 @@ def is_default(self) -> bool:
 
 def _verify_mount_root_dirs_are_unique(mounts: Union[None, Mount, List[Mount], Tuple[Mount]]) -> None:
     if isinstance(mounts, (list, tuple, set)):
-        root_dirs = [mount.root_dir for mount in mounts]
+        root_dirs = [mount.mount_path for mount in mounts]
```
Comment (lantiga) [inline src/lightning_app/utilities/packaging/cloud_compute.py:144]:
```suggestion
        mount_paths = [mount.mount_path for mount in mounts]
```

File: src/lightning_app/utilities/packaging/cloud_compute.py [hunk 3]
```diff
@@ -141,9 +141,9 @@ def is_default(self) -> bool:
...
         if len(set(root_dirs)) != len(root_dirs):
```
Comment (lantiga) [inline src/lightning_app/utilities/packaging/cloud_compute.py:145]:
```suggestion
        if len(set(mount_paths)) != len(mount_paths):
```

**Review by lantiga — CHANGES_REQUESTED** [review]
Looks good, very minor changes

File: src/lightning_app/storage/mount.py (hunk 1 above)
Comment (rlizzo) [reply src/lightning_app/storage/mount.py:49]: I already did! 

https://github.com/Lightning-AI/lightning/pull/15228/files#diff-6a0b6976b768aa8a7585702b656db550b2b1d997b8c4f765707d1dd2589cf986R40-R42

**Pull request merged**